In [1]:
from pymongo import MongoClient
import pandas as pd
import requests
import json
from dotenv import load_dotenv
import os
import pandas as pd
load_dotenv()
import api_calls as api

In [2]:
#We connect to our mongo database collection with the venues we extracted from 4square
client = MongoClient('localhost:27017')
db = client.get_database("ironhack")
geo = db.get_collection("Geobusinesses")

In [3]:
from pymongo import GEOSPHERE
geo.create_index([("geometry", GEOSPHERE)])

'geometry_2dsphere'

In [4]:
geo.find_one({})

{'_id': ObjectId('607d3f5755471685e3b839f7'),
 'name': 'Monte Carlo',
 'location': {'type': 'Point',
  'coordinates': [-122.0795899025, 37.3935957075]},
 'venue': 'nightclubs'}

In [5]:
# We choose a companies coordinates for every city in our analysis.

paypal_sanjose = [-121.9276, 37.3209]
paypal_sanjose_point = {"type": "Point",  "coordinates": paypal_sanjose}

google_mtview = [-122.084143, 37.421972]
google_mtview_point = {"type": "Point",  "coordinates": google_mtview}

yahoo_sunny = [-122.025485, 37.418531]
yahoo_sunny_point = {"type": "Point",  "coordinates": yahoo_sunny}

vmware_palo = [-122.145981, 37.403315] 
vmware_palo_point = {"type": "Point",  "coordinates": vmware_palo}

twitter_sanfran = [-122.416924, 37.776805]
twitter_sanfran_point = {"type": "Point",  "coordinates": twitter_sanfran}

In [6]:
#For us to consider something near, we will use a radius of 3kms, we employ the function near point, 
#that checks what is near each business location within our chosen radius
radius = 3000
near_paypal = api.near_point(paypal_sanjose_point, radius)
near_google = api.near_point(google_mtview_point, radius)
near_yahoo = api.near_point(yahoo_sunny_point, radius)
near_vmware = api.near_point(vmware_palo_point, radius)
near_twitter = api.near_point(twitter_sanfran_point, radius)

In [7]:
near_paypaldf = pd.DataFrame(near_paypal)
near_googledf = pd.DataFrame(near_google)
near_yahoodf = pd.DataFrame(near_yahoo)
near_vmwaredf = pd.DataFrame(near_vmware)
near_twitterdf = pd.DataFrame(near_twitter)

In [8]:
#As we can see there are a lot of duplicate values.
near_googledf

_id  \
0   607d4c5f55471685e3b83b49   
1   607d4c8a55471685e3b83bc8   
2   607d4c5f55471685e3b83b55   
3   607d418355471685e3b83a3b   
4   607d41b055471685e3b83ac7   
5   607d418355471685e3b83a3d   
6   607d418d55471685e3b83a6b   
7   607d4c5f55471685e3b83b66   
8   607d472455471685e3b83ad0   
9   607d473055471685e3b83ae4   
10  607d474b55471685e3b83b34   
11  607d418d55471685e3b83a5d   
12  607d418355471685e3b83a39   
13  607d41b055471685e3b83ab9   
14  607d4c5f55471685e3b83b4b   
15  607d4c8a55471685e3b83bd1   
16  607d4c5f55471685e3b83b4e   
17  607d4c8a55471685e3b83bdd   
18  607d4c6a55471685e3b83b7c   
19  607d418355471685e3b83a3a   
20  607d418d55471685e3b83a65   
21  607d41b055471685e3b83ab4   
22  607d4c5f55471685e3b83b4d   
23  607d4c6a55471685e3b83b7a   
24  607d4c5f55471685e3b83b4f   
25  607d4c6a55471685e3b83b7e   
26  607d4c8a55471685e3b83bd7   
27  607d4c5f55471685e3b83b51   
28  607d4c6a55471685e3b83b7b   
29  607d4c5f55471685e3b83b4a   
30  607d4c8a55471685e3b83bc1   
31  607d472455471685e3b83ad3   
32  607d474b55471685e3b83b33   
33  607d472455471685e3b83ad4   
34  607d473055471685e3b83ae8   
35  607d474b55471685e3b83b35   
36  607d4c5f55471685e3b83b52   
37  607d4c6a55471685e3b83b75   
38  607d4c5f55471685e3b83b4c   
39  607d4c8a55471685e3b83bc7   

                                                name  \
0                                    Zenith Tutoring   
1                                    Zenith Tutoring   
2      Chinese Church in Christ (CCiC) Mountain View   
3                                          Starbucks   
4                                          Starbucks   
5                                          Starbucks   
6                                          Starbucks   
7                           Crittenden Middle School   
8                            ASKA Drive & Fly by NFT   
9                            ASKA Drive & Fly by NFT   
10                           ASKA Drive & Fly by NFT   
11                                         Starbucks   
12                                         Starbucks   
13                                         Starbucks   
14                             Stevenson PACT School   
15                             Stevenson PACT School   
16                  Gideon Hausner Jewish Day School   
17                  Gideon Hausner Jewish Day School   
18                  Gideon Hausner Jewish Day School   
19                                         Starbucks   
20                                         Starbucks   
21                                         Starbucks   
22                              Acme Education group   
23                              Acme Education group   
24                                       Tutu School   
25                                       Tutu School   
26                                       Tutu School   
27            Computer Education at Abilities United   
28            Computer Education at Abilities United   
29   Carnegie Mellon West at NASA Ames Research Park   
30   Carnegie Mellon West at NASA Ames Research Park   
31                                             SOFIA   
32                                             SOFIA   
33                        Yellow Taxi Cab California   
34                        Yellow Taxi Cab California   
35                        Yellow Taxi Cab California   
36            Crescent Park Child Development Center   
37            Crescent Park Child Development Center   
38  Yew Chung International School of Silicon Valley   
39  Yew Chung International School of Silicon Valley   

                                             location      venue  
0   {'type': 'Point', 'coordinates': [-122.0787124...     school  
1   {'type': 'Point', 'coordinates': [-122.0787124...     school  
2   {'type': 'Point', 'coordinates': [-122.0901131...     school  
3   {'type': 'Point', 'coordinates': [-122.077594,...  starbucks  
4   {'type': 'Point', 'coordinates': [-122.077594,...  starbucks  
5   {'type': 

In [9]:
#We create a column that specifies if the value is duplicated by assigning True or False if it is.
near_googledf['duplicated'] = near_googledf.location.astype(str).duplicated(keep= 'first')
near_paypaldf['duplicated'] = near_paypaldf.location.astype(str).duplicated(keep= 'first')
near_yahoodf['duplicated'] = near_yahoodf.location.astype(str).duplicated(keep= 'first')
near_vmwaredf['duplicated'] = near_vmwaredf.location.astype(str).duplicated(keep= 'first')
near_twitterdf['duplicated'] = near_twitterdf.location.astype(str).duplicated(keep= 'first')

In [10]:
#We create a filter that picks up only the non duplicated venues, or the first value of the duplicated columns.
near_paypal_filtered = near_paypaldf[near_paypaldf['duplicated'] == False]
near_google_filtered = near_googledf[near_googledf['duplicated'] == False]
near_yahoo_filtered = near_yahoodf[near_yahoodf['duplicated'] == False]
near_vmwared_filtered = near_vmwaredf[near_vmwaredf['duplicated'] == False]
near_twitter_filtered = near_twitterdf[near_twitterdf['duplicated'] == False]

In [11]:
#We count the total values in our filtered dataframes and print them out
print(f'paypal has a total of: {len(near_paypal_filtered)} venues we are interested in')
print(f'google has a total of: {len(near_google_filtered)} venues we are interested in')
print(f'yahoo has a total of: {len(near_yahoo_filtered)} venues we are interested in')
print(f'vmware has a total of: {len(near_vmwared_filtered)} venues we are interested in')
print(f'twitter has a total of: {len(near_twitter_filtered)} venues we are interested in')

paypal has a total of: 31 venues we are interested in
google has a total of: 18 venues we are interested in
yahoo has a total of: 7 venues we are interested in
vmware has a total of: 23 venues we are interested in
twitter has a total of: 108 venues we are interested in


As we can see within a 3km radius:
1. Twitter(San Francisco) has 108 venues 
2. Paypal(San Jose) has 31
3. VmWare(Palo Alto) has 23
4. Google (Mt View) has 18
5. yahoo (Sunny Vale) has 7

From this analysis it is clear that San Francisco has the most places that would keep both management and our workers happy, but what is the best spot in San Francisco?
